issuetracker/src/test/java/org/openjdk/skara/issuetracker/IssueTrackerTests.java

```diff
@@ -127,7 +127,8 @@ void addIssueLink(TestInfo info) throws IOException {
 
             var links = issue2.links();
             assertEquals(1, links.size());
-            assertEquals(link, links.get(0));
+            assertEquals(link.relationship(), links.get(0).relationship());
```

Comment:
Here's an obvious drawback of the "var" syntax. What is this `links`? Why did not an ordinary equals suffice? Does this really relate to the rest of the changes, or is it an independent fix that happened to tag along?

Reply:
Yes, in modern intellij, I get grey type names for every `var` variable, which makes it really convenient. I too have to open most PRs I review in the IDE to properly read the code.

Regarding why equals won't suffice. Before this patch, `TestIssue::links` would just return a list of all `Link` objects that had been added to the Issue. In this patch, I had to change it to always refresh the target `Issue` of each link with a fresh snapshot, because that's what happens when you query for any real links. Another way to solve this would be to implement `Link::equals` to take this into account. That would probably be a good thing to do, but then this patch will touch product code. Long term, that's probably better and I can do that if you want.

Reply:
Looking closer at this, the `Link::equals` already does the right thing. The reason it's not sufficient is that `issue1` is being modified by setting a body. `TestIssue::setBody` just calls through to the `TestIssueStore` object. The snapshot in `issue1` doesn't change. When the link is created it points to this snapshot. When `issue2.links()` is called, the new logic will return a new `Link` object pointing to a fresh snapshot of `issue1`. If we just ran `equals` to compare `link` with `links.get(0)`, then it would return false, because `link` points to `issue1` without a body, while `links.get(0)` points to a fresh snapshot with a body.

Checking that `Issue::id` is equal is the correct thing to do here. Links point to an issue identified by an ID, not to a specific snapshot of the issue.